DataFormats/Headers/include/Headers/NameHeader.h

```diff
@@ -21,45 +21,61 @@
 #include <string>
 #include "Headers/DataHeader.h"
 
-namespace o2 {
-namespace header {
+namespace o2
+{
+namespace header
+{
 
 /// @struct NameHeader
 /// @brief an example data header containing a name of an object as a null terminated char arr.
 /// this is a template! at instantiation the template parameter determines the
 /// size of the held string array.
-/// a caveat with decoding is you have to use header::get<NameHeader<0>>(buffer)
+/// a caveat with decoding is (if length of sting is not known) you have to use header::get<NameHeader<0>>(buffer)
 /// to get it out of a buffer. May improve in the future if enough people complain.
+/// If the compiler decides to pad the end of struct, the length of padding is stored in the last byte
 /// @ingroup aliceo2_dataformats_dataheader
 template <size_t N>
 struct NameHeader : public BaseHeader {
   static const uint32_t sVersion;
   static const o2::header::HeaderType sHeaderType;
   static const o2::header::SerializationMethod sSerializationMethod;
-  NameHeader()
-  : BaseHeader(sizeof(NameHeader), sHeaderType, sSerializationMethod, sVersion)
-  , name()
+  NameHeader() : BaseHeader(sizeof(NameHeader), sHeaderType, sSerializationMethod, sVersion), name()
   {
-    memset(&name[0],'\0',N);
+    //set length of padding in the last byte
+    uint8_t* lastByte = reinterpret_cast<uint8_t*>(this) + sizeof(NameHeader) - 1;
+    *lastByte =
+      reinterpret_cast<const uint8_t*>(this) + sizeof(NameHeader) - reinterpret_cast<const uint8_t*>(&name[N]);
```

Comment:
isn't this `sizeof(NameHeader) - N`?

Reply:
I think not quite, there is still the content of the base class plus possible padding. Maybe it could be simplified, I don't know, I wanted to be explicit about which differences are calculated without relying on e.g. assumption that sizeof(char[9]) is 9.